### DISCUSSION TITLE
Not Serious, but a Thing with ActiveStorage

### DISCUSSION THREAD

**Discussion by resistorsoftware** [tilo/smarter_csv#207] [General]
If you open a file saved using ActiveStorage, you get a properly encoded UTF-8 file if the file was properly encoded UTF-8. If you then attack that open file with SmarterCSV, it throws this message:

WARNING: you are trying to process UTF-8 input, but did not open the input with "b:utf-8" option. See README file "NOTES about File Encodings".

I am pretty sure you cannot control how you open a file using ActiveStorage, as you just get the method "open" and not way to apply any features to that. So this message leads to a dead end. If anyone knows how to turn off this warning, that would be swell to know!

**Comment by resistorsoftware** [comment]
Just to be clear. Here is the code and warning:

 my_model.upload_file.open do |file|
  SmarterCSV.process(file, chunk_size: 100, file_encoding: "UTF-8") do
|chunk|

Ready to work on the file that was uploaded
09:57:38 worker.1 | WARNING: you are trying to process UTF-8 input, but did
not open the input with "b:utf-8" option. See README file "NOTES about File
Encodings".

**Comment by tilo** [comment]
that's a great question.. which `open` is it using? [this one?](https://apidock.com/rails/ActiveStorage/Blob/open) 🤔

**Comment by tilo** [comment]
@resistorsoftware can try to use this value for file_encoding: `b:utf-8`
